client/request.go

```diff
@@ -23,6 +23,9 @@ type Request interface {
 	// Request timeout
 	Timeout() time.Duration
 	SetTimeout(time.Duration)
+	// AccessToken for this request, used for authentication
+	AccessToken() string
+	SetAccessToken(string)
```

Comment:
Is this idiomatic? I just implemented `SetTimeout` in an emergency with the minimum effort possible. 

Once we have more other things to set (parent request information, trace id, etc) constructing a `request` will become a long series of `setX()` calls. Might be worth making a richer constructor (or exporting the fields because this isn't Java...)